resources/manager/registry.go

```diff
@@ -90,13 +89,13 @@ func (r *fetchersRegistry) ShouldRun(key string) bool {
 	return true
 }
 
-func (r *fetchersRegistry) Run(ctx context.Context, key string) ([]fetching.Resource, error) {
+func (r *fetchersRegistry) Run(ctx context.Context, key string, resCh chan<- fetching.ResourceInfo, cycleMetadata fetching.CycleMetadata) error {
 	registered, ok := r.reg[key]
 	if !ok {
-		return nil, fmt.Errorf("fetcher %v not found", key)
+		return fmt.Errorf("fetcher %v not found", key)
 	}
 
-	return registered.f.Fetch(ctx)
+	return registered.f.Fetch(ctx, resCh, cycleMetadata)
```

Comment:
What if the fetchers had an in channel, that would provide a new cycle metadata whenever a cycle should start?
Registering then will be actually registering the in channels of fetchers, and here the cycle metadata would have been sent on each one of the registered in channels.
This incorporates the pipeline flow further.

WDYT?

Maybe still worth splitting our modules to

1. Simple Synchronous modules with no in/out channels but simple interfaces
2. Pipeline wrappers that handle the pipeline work